src/common/pf_lldp.c

```diff
@@ -719,26 +749,100 @@ int pf_lldp_get_peer_station_name (
    int loc_port_num,
    pf_lldp_station_name_t * p_station_name)
 {
-   bool is_received;
+   pf_lldp_chassis_id_t chassis_id;
+   pf_lldp_port_id_t port_id;
+   char * p;
+   bool found = false;
+
    const pf_port_t * p_port_data = pf_port_get_state (net, loc_port_num);
+   memset (p_station_name, 0, sizeof (*p_station_name));
 
    os_mutex_lock (net->lldp_mutex);
+   port_id = p_port_data->lldp.peer_info.port_id;
+   chassis_id = p_port_data->lldp.peer_info.chassis_id;
+   os_mutex_unlock (net->lldp_mutex);
+
+   if (port_id.is_valid)
+   {
+      p = strchr (port_id.string, '.');
+      if (p == NULL)
+      {
+         /* PF_LLDP_NAME_OF_DEVICE_MODE_STANDARD
+          * Example: port-001.dut
+          */
+         snprintf (
+            p_station_name->string,
+            sizeof (p_station_name->string),
+            "%s",
+            p + 1);
+         found = true;
+      }
+   }
+
+   if (!found && chassis_id.is_valid)
+   {
+      /* PF_LLDP_NAME_OF_DEVICE_MODE_LEGACY */
+      if (chassis_id.subtype != PF_LLDP_SUBTYPE_MAC)
+      {
+         snprintf (
+            p_station_name->string,
+            sizeof (p_station_name->string),
+            "%s",
+            chassis_id.string);
+      }
+      else
+      {
+         pf_lldp_mac_to_str (
+            (uint8_t *)chassis_id.string,
+            p_station_name->string,
+            sizeof (p_station_name->string));
+      }
+      found = true;
+   }
+
+   /* Do not use peer mac address here since its status is unknown. */
 
-   /* Assume that peer's NameOfStation is the same as its ChassisId.
-    * TODO: Find out the correct way to choose what value to return.
-    * Value should be part of either received Chassis ID or Port ID.
-    */
-   memset (p_station_name, 0, sizeof (*p_station_name));
-   snprintf (
-      p_station_name->string,
-      sizeof (p_station_name->string),
-      "%s",
-      p_port_data->lldp.peer_info.chassis_id.string);
    p_station_name->len = strlen (p_station_name->string);
-   is_received = p_port_data->lldp.peer_info.chassis_id.is_valid;
 
+   return p_station_name->len ? 0 : -1;
+}
+
+int pf_lldp_get_peer_port_name (
+   pnet_t * net,
+   int loc_port_num,
+   pf_lldp_port_name_t * p_port_name)
+{
+   pf_lldp_port_id_t port_id;
+   uint16_t i;
+   const pf_port_t * p_port_data = pf_port_get_state (net, loc_port_num);
+   memset (p_port_name, 0, sizeof (*p_port_name));
+
+   os_mutex_lock (net->lldp_mutex);
+   port_id = p_port_data->lldp.peer_info.port_id;
    os_mutex_unlock (net->lldp_mutex);
-   return is_received ? 0 : -1;
+
+   if (port_id.is_valid)
+   {
+      for (i = 0; (i < port_id.len) && (port_id.string[i] != '\0') &&
+                  (i < (sizeof (p_port_name->string) - 1));
```

Comment:
Why was the condition `i < (sizeof (p_port_name->string) - 1` introduced?
Given that `i < port_id.len`, this should always be true

Reply:
port_id.string is 256 characters. I think I risk writing outside port_name->string if  the port_id string is not formatted as it should be.

Reply:
Sorry, I misread it as `sizeof (port_id->string)` instead of `sizeof (p_port_name->string)`.